File: Python/SalesByMatch.py
# https://www.hackerrank.com/challenges/sock-merchant/problem

class solution:
    def sockMerchant(n, ar):
        reult = 0
        map = {}
        
        for color in ar:
            map[color] = map.get(color, 0) + 1
            
        for value in map.values():
            result += value // 2
            
        return result
    
    if __name__ == "__main__":
        n = int(input().strip())
        ar = list(map(int, input().strip().split()))
        result = solution.sockMerchant(n, ar)
        print(result)

File: SalesByMatch.java
// https://www.hackerrank.com/challenges/sock-merchant/problem?isFullScreen=true

import java.util.HashMap;
import java.util.List;

public class SalesByMatch {
    public static int sockMerchant(int n, List<Integer> ar) {
        int result = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int num : ar){
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        for(int value : map.values()){
            result = result + (value / 2);
        }
        return result;
    }

    public static void main(String[] args) {
        SalesByMatch solution = new SalesByMatch();
        List<Integer> ar = List.of(10, 20, 20, 10, 10, 30, 50, 10, 20);
        int n = ar.size();
        int result = solution.sockMerchant(n, ar);
        System.out.println(result);
    }
}
